### ISSUE TITLE
Copying a running container on Ceph should not be allowed

### ISSUE THREAD

**Issue by escabo** [canonical/lxd#12658]
# Required information

 * Distribution: snap
 * Distribution version: 5.19

# Issue description

When trying to copy a running container on zfs or btrfs, we get this message:
```
Error: Unable to perform live container migration. CRIU isn't installed on the source server. To migrate the container, stop the container before migration or install CRIU on the source serve
```
The copy proceeds without any error when the storage backend is Ceph.

# Steps to reproduce

lxc launch ubuntu:jammy c2 --storage remote
lxc copy c2 c2copy

**Comment by roosterfish** [comment]
Looks like Ceph backed VM's don't get frozen too during copy. Just tested this on `edge`.

**Comment by tomponline** [comment]
@roosterfish are you working on this? I'll re-assign to you. Please self-assign if you start working on an issue so we know. Ta

**Comment by tomponline** [comment]
> Looks like Ceph backed VM's don't get frozen too during copy. Just tested this on `edge`.

I wouldn't expect them to be frozen during the copy, but i would expect a temporary snapshot to be taken and that the VM should potentially be frozen during the snapshot being taken. But that depends on the target pool type.

**Comment by roosterfish** [comment]
Not particularly, I have just ran across it since I have added https://github.com/canonical/lxd-ci/pull/41 which copies a running container and I was wondering what happens for VMs.

**Comment by tomponline** [comment]
OK ill reassign to din

**Comment by tomponline** [comment]
@MusicDin I'm going to give this one to Pedro

**Comment by escabo** [comment]
Do we expect different behaviors if the copy is from to different nodes vs on the same node?

**Comment by tomponline** [comment]
No, live migration for containers cannot occur without CRIU (not that it will work in majority of cases with CRIU anyway, but thats out of scope for this issue).

**Comment by escabo** [comment]
So we should prevent it on other storage drivers as well as the current behavior, for ZFS for instance, is different if the copy is on the same node vs from node1 to node2.

**Comment by hamistao** [comment]
It is not a matrix but I think this covers all the important behavior:

local copies succeed both in the local and remote storage cases.
If copying a container using local storage to a different node, the error occurs.
If copying a container using remote storage(ceph) to a different node, it succeeds and the copy works normally.

lt also caught my attention that the CLI assumes the copy is on a remote node if the storage is local and no target is specified.
When the container storage is remote, it assumes the copy is local, but if a remote target is specified the copy suceeds anyway.
